packages/fiori/src/DynamicPage.ts

```diff
@@ -0,0 +1,338 @@
+	footer!: HTMLElement[];
+
+	onEnterDOM() {
+		ResizeHandler.register(this, this._updateMediaRange);
+	}
+
+	onExitDOM() {
+		ResizeHandler.deregister(this, this._updateMediaRange);
+	}
+
+	onBeforeRendering() {
+		if (this.dynamicPageTitle) {
+			this.dynamicPageTitle.snapped = this.headerSnapped;
+		}
+	}
+
+	get classes() {
+		return {
+			root: {
+				"ui5-dynamic-page-root": true,
+			},
+			scrollContainer: {
+				"ui5-dynamic-page-scroll-container": true,
+			},
+			headerWrapper: {
+				"ui5-dynamic-page-title-header-wrapper": true,
+			},
+			content: {
+				"ui5-dynamic-page-content": true,
+			},
+			fitContent: {
+				"ui5-dynamic-page-fit-content": true,
+			},
+			footer: {
+				"ui5-dynamic-page-footer": true,
+			},
+			spacer: {
+				"ui5-dynamic-page-spacer": true,
+			},
+		};
+	}
+
+	get dynamicPageTitle(): DynamicPageTitle | null {
+		return this.querySelector<DynamicPageTitle>("[ui5-dynamic-page-title]");
+	}
+
+	get dynamicPageHeader(): DynamicPageHeader | null {
+		return this.querySelector<DynamicPageHeader>("[ui5-dynamic-page-header]");
+	}
+
+	get scrollContainer(): HTMLElement | null {
+		return this.shadowRoot!.querySelector<HTMLElement>(".ui5-dynamic-page-scroll-container");
+	}
+
+	get headerActions(): DynamicPageHeaderActions | null {
+		return this.shadowRoot!.querySelector<DynamicPageHeaderActions>("ui5-dynamic-page-header-actions");
+	}
+
+	get actionsInTitle(): boolean {
+		return this.headerSnapped || this.showHeaderInStickArea || this.headerPinned;
+	}
+
+	get headerInTitle(): boolean {
+		return !this.headerSnapped && (this.showHeaderInStickArea || this.headerPinned);
+	}
+
+	get headerInContent(): boolean {
+		return !this.headerSnapped && !this.headerInTitle;
+	}
+
+	get _headerLabel() {
+		return this.headerSnapped
+			? DynamicPage.i18nBundle.getText(DYNAMIC_PAGE_ARIA_LABEL_SNAPPED_HEADER)
+			: DynamicPage.i18nBundle.getText(DYNAMIC_PAGE_ARIA_LABEL_EXPANDED_HEADER);
+	}
+
+	get _headerExpanded() {
+		return !this.headerSnapped;
+	}
+
+	get _accAttributesForHeaderActions() {
+		return {
+			controls: `${this._id}-header`,
+		};
+	}
+
+	snapOnScroll() {
+		debounce(() => this.snapTitleByScroll(), SCROLL_DEBOUNCE_RATE);
+	}
+
+	snapTitleByScroll() {
+		if (!this.dynamicPageTitle || !this.dynamicPageHeader || this.headerPinned) {
+			return;
+		}
+
+		const scrollTop = this.scrollContainer!.scrollTop;
+
+		if (this.skipSnapOnScroll) {
+			this.skipSnapOnScroll = false;
+			return;
+		}
+
+		if (scrollTop > this.dynamicPageHeader.getBoundingClientRect().height) {
+			this.headerSnapped = true;
+			this.showHeaderInStickArea = false;
+		} else {
+			this.headerSnapped = false;
+		}
+
+		this.dynamicPageTitle.snapped = this.headerSnapped;
+	}
+
+	async onExpandClick() {
+		this._toggleHeader();
+		await renderFinished();
+		this.headerActions?.focusExpandButton();
+		announce(this._headerLabel, InvisibleMessageMode.Polite);
+	}
+
+	async onPinClick() {
+		this.headerPinned = !this.headerPinned;
```

Comment:
headerPinned, headerSnapped are public properties that we internally change due to user interaction. I think we need to fire events to notify for these changes, and this is how we usually handle this case in other components.